modoboa/core/api/v2/views.py

```diff
@@ -70,3 +82,117 @@ def post(self, request, *args, **kwargs):
 
         return response.Response(
             serializer.validated_data, status=status.HTTP_200_OK)
+
+
+class RestPasswordResetView(APIView):
+    """
+    An Api View which provides a method to request a password reset token based on an e-mail address
+    """
+
+    throttle_classes = ()
+    permission_classes = ()
+    serializer_class = PasswordRecoverySerializer
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        email = serializer.validated_data["email"]
+
+        if len(get_user_model()._default_manager.filter(
+                email__iexact=email, is_active=True)
+               .exclude(Q(secondary_email__isnull=True) | Q(secondary_email=""))) == 0:
+            return response.Response({"Status": "no_user"}, status=404)
+
+        form = PasswordResetForm(data={"email": email})
```

Comment:
Why not use a serializer instead?

Reply:
So we don't have to rewrite what django has already done regarding password reset (token creation, verification, etc.). Would it be better to rewrite it into a serializer ?